lidar-plot-base-transformer/transformer.py

```diff
@@ -0,0 +1,882 @@
+            key_terms: the keys to look for
+        Returns:
+            Returns the found value or an empty string
+        """
+        for one_key in key_terms:
+            value = __internal__.recursive_metadata_search(metadata_list, one_key)
+            if value:
+                return value
+
+        return ''
+
+    @staticmethod
+    def prepare_algorithm_metadata() -> tuple:
+        """Prepares metadata with algorithm information
+        Return:
+            Returns a tuple with the name of the algorithm and a dictionary with information on the algorithm
+        """
+        return (__internal__.get_algorithm_definition_str('ALGORITHM_NAME', 'unknown'),
+               {
+                    'version': __internal__.get_algorithm_definition_str('VERSION', 'x.y'),
+                    'traits': __internal__.get_algorithm_definition_str('VARIABLE_NAMES', ''),
+                    'units': __internal__.get_algorithm_definition_str('VARIABLE_UNITS', ''),
+                    'labels': __internal__.get_algorithm_definition_str('VARIABLE_LABELS', '')
+               })
+
+    @staticmethod
+    def image_get_geobounds(filename: str) -> list:
+        """Uses gdal functionality to retrieve rectilinear boundaries from the file
+
+        Args:
+            filename: path of the file to get the boundaries from
+
+        Returns:
+            The upper-left and calculated lower-right boundaries of the image in a list upon success.
+            The values are returned in following order: min_y, max_y, min_x, max_x. A list of numpy.nan
+            is returned if the boundaries can't be determined
+        """
+        try:
+            src = gdal.Open(filename)
+            ulx, xres, _, uly, _, yres = src.GetGeoTransform()
+            lrx = ulx + (src.RasterXSize * xres)
+            lry = uly + (src.RasterYSize * yres)
+
+            min_y = min(uly, lry)
+            max_y = max(uly, lry)
+            min_x = min(ulx, lrx)
+            max_x = max(ulx, lrx)
+
+            return [min_y, max_y, min_x, max_x]
+        except Exception as ex:
+            logging.exception("[image_get_geobounds] Exception caught processing file: %s", filename)
+
+        return [np.nan, np.nan, np.nan, np.nan]
+
+    @staticmethod
+    def get_epsg(filename):
+        """Returns the EPSG of the geo-referenced image file
+        Args:
+            filename(str): path of the file to retrieve the EPSG code from
+        Return:
+            Returns the found EPSG code, or None if it's not found or an error ocurred
+        """
+        try:
+            src = gdal.Open(filename)
+
+            proj = osr.SpatialReference(wkt=src.GetProjection())
+
+            return proj.GetAttrValue('AUTHORITY', 1)
+        except Exception as ex:
+            logging.exception("[get_epsg] Exception caught processing file: %s", filename)
+
+        return None
+
+    @staticmethod
+    def get_centroid_latlon(filename: str):
```

Comment:
This won't work for LAS files. See https://github.com/AgPipeline/transformer-plotclip/blob/ca48c9805d555bb42192bab8b682a361a542b5b5/transformer.py#L69

Reply:
Ooh cool thank you, I'll take a look at it.

Reply:
In that same file it has a `get_las_epsg_from_header` function where this file has just the `get_espg` function, should be looking at that as well?

Reply:
yeah, this function calls `get_las_espg_from_header` so I'll have a lot of restructuring to do.